app/features/receive/receive-cashu-token.tsx

```diff
@@ -22,29 +29,67 @@ type Props = {
 export default function ReceiveToken({ token }: Props) {
   const [_, copyToClipboard] = useCopyToClipboard();
   const { toast } = useToast();
-  const {
-    data: tokenData,
-    isClaiming,
-    handleClaim,
-    setReceiveAccount,
-  } = useReceiveCashuToken({
+  const addCashuAccount = useAddCashuAccount();
+  const { claimableToken, cannotClaimReason } = useGetClaimableToken({
     token,
-    cashuPubKey:
+    cashuPubKey: // TODO: replace with user's pubkey from OS
       '038127ae202c95f4cd4ea8ba34e73618f578adf516db553a902a8589796bdc373',
   });
-
+  const sourceAccount = useGetCashuTokenSourceAccount(token);
   const {
-    receiveAccount,
     selectableAccounts,
-    crossMintSwapDisabled,
+    receiveAccount,
     receiveAccountIsSource,
-    cannotClaimReason,
-    isMintKnown,
-    claimableToken,
-  } = tokenData;
+    isSourceAccountAdded,
+    isCrossMintSwapDisabled,
+    setReceiveAccount,
+  } = useReceiveCashuTokenAccounts(sourceAccount);
+
+  const { status, startClaimingTokenToAccount } = useReceiveCashuToken({
+    onError: (error) => {
+      toast({
+        title: 'Failed to claim token',
+        description: error.message,
+        variant: 'destructive',
+      });
+    },
+  });
+
+  const handleClaim = async () => {
+    if (!claimableToken) {
+      return;
+    }
+
+    // QUESTION: when we call startReceive the account needs to be created already.
+    // Is there a better way to do this?
+    let account: CashuAccount = receiveAccount;
+
+    if (!isSourceAccountAdded && receiveAccountIsSource) {
```

Comment:
I think this can be simplified. It doesn't matter what the source account is at this point. Only thing that matters is if the account selected to receive to is saved or not so we should change to code to reflect that